File: backend/src/models/Whatsapp.ts
import {
  Table,
  Column,
  CreatedAt,
  UpdatedAt,
  Model,
  DataType,
  PrimaryKey,
  AutoIncrement,
  Default,
  AllowNull,
  HasMany,
  Unique,
  BelongsToMany,
  ForeignKey,
  BelongsTo
} from "sequelize-typescript";
import Tenant from "./Tenant";
import Queue from "./Queue";
import Ticket from "./Ticket";
import WhatsappQueue from "./WhatsappQueue";

@Table
class Whatsapp extends Model<Whatsapp> {
  @PrimaryKey
  @AutoIncrement
  @Column
  id: number;

  @AllowNull
  @Unique
  @Column(DataType.TEXT)
  name: string;

  @Column(DataType.TEXT)
  session: string;

  @Column(DataType.TEXT)
  qrcode: string;

  @Column(DataType.STRING)
  number: string;

  @Column(DataType.STRING)
  profilePicUrl: string;

  @Column
  status: string;

  @Column
  battery: string;

  @Column
  plugged: boolean;

  @Column
  retries: number;

  @Column(DataType.TEXT)
  greetingMessage: string;

  @Column(DataType.TEXT)
  farewellMessage: string;

  // Campos opcionais usados por módulos mais novos (compatibilidade de build)
  type?: string;
  chatConfig?: any;

  @Default(false)
  @AllowNull
  @Column
  isDefault: boolean;

  @Column(DataType.BOOLEAN)
  syncHistory: boolean;

  @Column(DataType.TEXT)
  syncPeriod: string;

  @Default(false)
  @AllowNull
  @Column(DataType.BOOLEAN)
  keepAlive: boolean;

  @CreatedAt
  createdAt: Date;

  @UpdatedAt
  updatedAt: Date;

  @HasMany(() => Ticket)
  tickets: Ticket[];

  @BelongsToMany(() => Queue, () => WhatsappQueue)
  queues: Array<Queue & { WhatsappQueue: WhatsappQueue }>;

  @HasMany(() => WhatsappQueue)
  whatsappQueues: WhatsappQueue[];

  @ForeignKey(() => Tenant)
  @Column(DataType.UUID)
  tenantId: number | string;

  @BelongsTo(() => Tenant)
  tenant: Tenant;
}

export default Whatsapp;

File: bussines/internal/models/whatsapp.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Whatsapp struct {
	ID              int       `gorm:"primaryKey" json:"id"`
	Session         string    `json:"session"`
	Qrcode          string    `gorm:"type:text" json:"qrcode"`
	Status          string    `json:"status"`
	Battery         string    `json:"battery"`
	Plugged         bool      `json:"plugged"`
	Name            string    `gorm:"unique;not null" json:"name"`
	IsDefault       bool      `gorm:"column:isDefault;not null;default:false" json:"isDefault"`
	Retries         int       `gorm:"not null;default:0" json:"retries"`
	GreetingMessage string    `gorm:"column:greetingMessage" json:"greetingMessage"`
	FarewellMessage string    `gorm:"column:farewellMessage" json:"farewellMessage"`
	TenantID        uuid.UUID `gorm:"column:tenantId;type:uuid" json:"tenantId"`
	SyncHistory     bool      `gorm:"column:syncHistory;default:false" json:"syncHistory"`
	SyncPeriod      string    `gorm:"column:syncPeriod" json:"syncPeriod"`
	Number          string    `json:"number"`
	ProfilePicUrl   string    `gorm:"column:profilePicUrl" json:"profilePicUrl"`
	KeepAlive       bool      `gorm:"column:keepAlive;default:false" json:"keepAlive"`
	CreatedAt       time.Time `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt       time.Time `gorm:"column:updatedAt" json:"updatedAt"`

	// Relations
	Tickets []Ticket `gorm:"foreignKey:WhatsappID" json:"tickets,omitempty"`
}

func (Whatsapp) TableName() string {
	return "Whatsapps"
}
